tools/codegen/dest/lazy_ir.py

```diff
@@ -0,0 +1,138 @@
+from typing import List, Optional, Union, Tuple
+import itertools
+from typing_extensions import Literal
+from dataclasses import dataclass
+import textwrap
+from tools.codegen import local
+from tools.codegen.context import method_with_native_function, native_function_manager
+from tools.codegen.utils import Target, mapMaybe
+from tools.codegen.model import (BaseType, OptionalType, DispatchKey, NativeFunction,
+                                 NativeFunctionsGroup, SchemaKind, FunctionSchema,
+                                 TensorOptionsArguments, ListType,
+                                 DeviceCheckType, Argument, assert_never,
+                                 is_cuda_dispatch_key, BackendIndex,
+                                 gets_generated_out_inplace_wrapper)
+from tools.codegen.api.types import (BaseTy, BaseCppType, BaseCType, OptionalCType,
+                                     Binding, ConstRefCType, NamedCType,
+                                     CppSignature, CppSignatureGroup,
+                                     Expr, MutRefCType, kernel_signature,
+                                     DispatcherSignature)
+import tools.codegen.api.meta as meta
+import tools.codegen.api.cpp as cpp
+import tools.codegen.api.structured as structured
+from tools.codegen.api.translate import translate
+from tools.codegen.selective_build.selector import SelectiveBuilder
+
+# Generates {backend}_lazy_ir.h and .cpp
+#
+valueListT = BaseCppType('at', 'ValueList') # TODO, not sure this type exists
+valueT = BaseCppType('ir', 'Value')
```

Comment:
A thought: in the old code structure in XLA, we generate functions which take Value/ValueList as arguments and then call into them. I'm not sure this is really necessary: couldn't we directly pass Tensors into the generated IR construction code, and just have the IR construction convert the Tensor into a Value? Then you wouldn't need `process_ir_types` below.

Reply:
It could still be a convenience, to put the conversion logic into the base 'node' class so that we don't have to codegen the conversions.  But yea, I guess there is no way around generating the correct 'Value' type in some part of the generated code.